src/main/proto/ga4gh/read_service.proto

```diff
@@ -91,6 +99,52 @@ message GetReadGroupSetRequest {
   string read_group_set_id = 1;
 }
 
+// ******************  /readgroups  *********************/
+
+// This request maps to the body of `POST /readgroups/search` as JSON.
+
+
+message SearchReadGroupsRequest {
+  // The dataset to search.
+  string dataset_id = 1;
+  
+  // Only return readgroups within the provided read group set ID.
+  string read_group_set_id = 2;
+
+  // Only return read groups with this name (case-sensitive, exact match).
+  string name = 3;
+
+  //  Specifying the id of a BioSample record will return only readgroups
+  //  with the given bioSampleId.
+  string bio_sample_id = 4;
```

Comment:
Well, that brings up some old memories :)  So there are too many examples to pick from, and below are a few snippets with their respective links, which go into more detail:

> @delagoya posted on https://github.com/ga4gh/schemas/issues/32#issuecomment-41669970:
* A `ReadGroup` is defined on submission by the submitter and represents an atomic grouping of reads (could be library, could barcode, could be source sample). A `Read` can only belong to one `ReadGroup`. 
* `ReadSet` is an arbitrary collection of Reads that represents a set of reads to be analyzed at the same time. 

> @birney posted on https://github.com/ga4gh/schemas/issues/32#issuecomment-41779223:
ReadGroups - these are sets of reads which should be analysed together because the scientist who generated the data say that they are together. 
Samples - A ReadGroup has one and only one Sample. However we can either allow Samples to have > 1 ReadGroup and/or allow Samples to be equivalent in some testing manner as for sure there will be >1 ReadGroup per Sample in a number of scenarios.
ReadGroupLists - these are sets of ReadGroups allowing functions to call across ReadGroups.

> @lh3 posted on https://github.com/ga4gh/schemas/issues/24#issue-31877179:
* A read group is the smallest set of homogeneous reads. Although there is still ambiguity in this definition, ReadGroup is in practice more clearly defined and more stable than ReadSet (see #16) and DataSet. It is a first-class concept in SAM widely used in data processing. Many downstream tools, such as GATK, samtools and Picard, focus on read groups and frequently ignore the concept of file (or ReadSet). As a matter of fact, all ReadGroups from 1000g are associated with SRA/ERA accession numbers. Most 1000g BAM files are not for now, because ReadSet is harder to define than ReadGroup.
* Sample and experimental info are attached to ReadGroup in SAM, not to a physical file or a ReadSet in the schema. When we query by sample, by library or by sequencing technology, we are directly searching for ReadGroups, not for ReadSets. In addition, it is possible that a ReadSet may contain ReadGroups from different samples - SAM is the same. When we query by sample, we will have to break a ReadSet. ReadGroup is the minimal unit.

Hope it helps,
~p